cse-cc-import-runner-app/src/main/java/com/farao_community/farao/cse/import_runner/app/dichotomy/DichotomyResultHelper.java

```diff
@@ -36,15 +39,16 @@ public DichotomyResultHelper(FileImporter fileImporter) {
         this.fileImporter = fileImporter;
     }
 
-    public String getLimitingElement(DichotomyResult<DichotomyRaoResponse> dichotomyResult) throws IOException {
+    public String getLimitingElement(DichotomyResult<DichotomyRaoResponse> dichotomyResult) {
         DichotomyStepResult<DichotomyRaoResponse> highestValidStepResult = dichotomyResult.getHighestValidStep();
         double worstMargin = Double.MAX_VALUE;
         Optional<FlowCnec> worstCnec = Optional.empty();
         Crac crac = fileImporter.importCracFromJson(highestValidStepResult.getValidationData()
             .getRaoResponse().getCracFileUrl());
         RaoResult raoResult = fileImporter.importRaoResult(highestValidStepResult.getValidationData()
             .getRaoResponse().getRaoResultFileUrl(), crac);
-        for (FlowCnec flowCnec : crac.getFlowCnecs()) {
+        Set<FlowCnec> optimizedFlowCnecs = crac.getFlowCnecs().stream().filter(Cnec::isOptimized).collect(Collectors.toSet());
+        for (FlowCnec flowCnec : optimizedFlowCnecs) {
             double margin = computeFlowMargin(raoResult, flowCnec);
```

Comment:
Then use this newly created method here as well, so we're sure the method to find the limiting element is always the same